rust/perspective-viewer/src/rust/components/column_settings_sidebar.rs

```diff
@@ -0,0 +1,128 @@
+// ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+// ┃ ██████ ██████ ██████       █      █      █      █      █ █▄  ▀███ █       ┃
+// ┃ ▄▄▄▄▄█ █▄▄▄▄▄ ▄▄▄▄▄█  ▀▀▀▀▀█▀▀▀▀▀ █ ▀▀▀▀▀█ ████████▌▐███ ███▄  ▀█ █ ▀▀▀▀▀ ┃
+// ┃ █▀▀▀▀▀ █▀▀▀▀▀ █▀██▀▀ ▄▄▄▄▄ █ ▄▄▄▄▄█ ▄▄▄▄▄█ ████████▌▐███ █████▄   █ ▄▄▄▄▄ ┃
+// ┃ █      ██████ █  ▀█▄       █ ██████      █      ███▌▐███ ███████▄ █       ┃
+// ┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┫
+// ┃ Copyright (c) 2017, the Perspective Authors.                              ┃
+// ┃ ╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌ ┃
+// ┃ This file is part of the Perspective library, distributed under the terms ┃
+// ┃ of the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0). ┃
+// ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+mod attributes_tab;
+mod style_tab;
+
+use std::fmt::Display;
+
+use yew::{function_component, html, Callback, Html, Properties};
+
+use super::containers::tablist::Tab;
+use super::viewer::ColumnLocator;
+use crate::components::column_settings_sidebar::attributes_tab::AttributesTab;
+use crate::components::column_settings_sidebar::style_tab::StyleTab;
+use crate::components::containers::sidebar::Sidebar;
+use crate::components::containers::tablist::TabList;
+use crate::components::style::LocalStyle;
+use crate::presentation::Presentation;
+use crate::renderer::Renderer;
+use crate::session::Session;
+use crate::{clone, css, html_template};
+
+#[derive(Debug, Default, Clone, PartialEq)]
+pub enum ColumnSettingsTab {
+    #[default]
+    Attributes,
+    Style,
+}
+impl Display for ColumnSettingsTab {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!("{self:?}"))
+    }
+}
+impl Tab for ColumnSettingsTab {}
+
+#[derive(PartialEq, Clone, Properties)]
+pub struct ColumnSettingsProps {
+    pub selected_column: ColumnLocator,
+    pub on_close: Callback<()>,
+    pub session: Session,
+    pub renderer: Renderer,
+    pub presentation: Presentation,
+}
+
+#[function_component]
+pub fn ColumnSettingsSidebar(p: &ColumnSettingsProps) -> Html {
+    let column_name = match p.selected_column.clone() {
+        ColumnLocator::Expr(maybe_name) => maybe_name.unwrap_or("New Column".to_string()),
```

Comment:
This should be an abort - returning `"New Column"` shouldn't ever be reached but if it does, this error condition is just kicked down the road here where it may be difficult to debug.

Reply:
ColumnLocator::Expr(None) is used when creating a new Expression column. However, we shouldn't be hardcoding the name, so I used the same function as in the expression_editor to generate the name.